BoxContentSDK/BoxContentSDK/BOXNSURLSessionDelegate.m

```diff
@@ -0,0 +1,303 @@
+//
+//  BOXNSURLSessionDelegate.m
+//  BoxContentSDK
+//
+//  Created by Thuy Nguyen on 12/15/16.
+//  Copyright © 2016 Box. All rights reserved.
+//
+
+#import "BOXNSURLSessionDelegate.h"
+
+/**
+    FIXME: implement details for the callbacks of interfaces
+     NSURLSessionDelegate
+     NSURLSessionTaskDelegate
+     NSURLSessionDataDelegate
+     NSURLSessionDownloadDelegate
+     NSURLSessionStreamDelegate
+ */
+
+
+@implementation BOXNSURLSessionDelegate
+
+/*
+ * Messages related to the URL session as a whole
+ */
+#pragma mark - NSURLSessionDelegate
+
+/* The last message a session receives.  A session will only become
+ * invalid because of a systemic error or when it has been
+ * explicitly invalidated, in which case the error parameter will be nil.
+ */
+- (void)URLSession:(NSURLSession *)session didBecomeInvalidWithError:(nullable NSError *)error
+{
+
+}
+
+/* If implemented, when a connection level authentication challenge
+ * has occurred, this delegate will be given the opportunity to
+ * provide authentication credentials to the underlying
+ * connection. Some types of authentication will apply to more than
+ * one request on a given connection to a server (SSL Server Trust
+ * challenges).  If this delegate message is not implemented, the
+ * behavior will be to use the default handling, which may involve user
+ * interaction.
+ */
+- (void)URLSession:(NSURLSession *)session didReceiveChallenge:(NSURLAuthenticationChallenge *)challenge
+ completionHandler:(void (^)(NSURLSessionAuthChallengeDisposition disposition, NSURLCredential * _Nullable credential))completionHandler
+{
+
```

Comment:
as discussed offline, remove the empty implementations of those delegate methods to be added selectively later as needed.